src/lxc/utils.c

```diff
@@ -2246,3 +2246,61 @@ int lxc_unstack_mountpoint(const char *path, bool lazy)
 
 	return umounts;
 }
+
+int run_command(char *buf, size_t buf_size, int (*child_fn)(void *), void *args)
+{
+	pid_t child;
+	int ret, fret, pipefd[2];
+	ssize_t bytes;
+
```

Comment:
Best to do a 
 if (buf_size > 0 && buf) buf[0] = '\0';

because note that right now if pipe() fails here, lxc_map_ids() will print the un-initialized buffer out in its error message.